### DISCUSSION TITLE
Sending email receipt with info from POS store cart contents & customer details

### DISCUSSION THREAD

**Discussion by c3piio** [btcpayserver/btcpayserver#4766] [Ideas / Feature Requests]
Hello, I am configuring emails for a store with a POS app. The POS app is shop + cart, so multiple items can be purchased. Customer details (emails + address) are collected during checkout.

- Can customer data such as name and address be included in the email via placeholders? The only supported placeholders I could find are seen here: https://docs.btcpayserver.org/Notifications/#email-rules
- Can cart contents be included in the email with similar placeholders? For example, item, quantity and price.

- On the invoice and receipt page, is it possible to include cart contents and customer data as well? This is just a question, since there are privacy issues with this. However, having the information included in the emails would solve all problems for managing orders and giving customers a proper receipt.

EDIT:

I would suggest placeholders for:

- customer email, address, or essentially all form data
- creating placeholders for custom form data (such as phone number)
- placeholders for cart contents
- placeholders for order details such as date of purchase, and today's date

This would allow to easily create a complete receipt that the customer can use when needing to present a proof of purchase for a variety of reasons, such as warranty.

**Comment by Liongrass** [comment]
I thought about the same thing yesterday, I would like to give customers a copy of their receipt automatically. I'm processing orders for pickup, and I fear customers won't remember exactly what they ordered, and this will result in extra work for me (replying to customer emails) and confusion at pickup.

I have Email configured and am sending out a purchase confirmation. Sending out the receipt as part of that would be great.

**Reply by pavlenex** [reply]
@Kukks afair there's an easy way to send out a receipt, what exactly needs to be included? Is it [this](https://docs.btcpayserver.org/Forms/#showing-user-input-on-the-receipt)?

**Reply by dennisreimann** [reply]
@pavlenex Nope, that's for the admin backend. 

I think we could add a placeholder for the receipt link to the emails — for now this could be achieved by assembling the link yourself usingh the `{Invoice.Id}` placeholder: You should be able to add `https://mybtcpay.com/i/{Invoice.Id}/receipt` to the email that gets sent to the customer.

**Reply by Liongrass** [reply]
Hm, I tried adding `https://mybtcpay.com/i/{InvoiceId}/receipt` in a hyperlink, but it doesnt work, it remains exactly that. When adding it as raw text, it becomes `https://btcpay.containerbrewing.ca/i/%7BInvoiceId%7D/receipt` I did add {InvoiceID} elswhere in the email, and it does properly display the invoice ID. Do I have to escape some characters?

**Reply by c3piio** [reply]
I also solved this in the end by simply including a link to the receipt, instead of emailing the receipt contents to the customer. It works well.

**Reply by Liongrass** [reply]
Thanks @c3piio ! How are you adding a link to the receipt?

**Comment by c3piio** [comment]
I added it as a hyperlink with the invoice placeholder as described. So for
example a link called receipt, with the URL pointing to the receipt with
the invoice placeholder as described above.

**Reply by Liongrass** [reply]
Would you be willing to share the exact html code that you are using? As you can see above I wasnt able to get it to work

**Reply by c3piio** [reply]
I used this: https://mybtcpay.com/i/{InvoiceId}/receipt Last time when I tested sending emails with this link, it worked. I haven't tried recently however.

**Reply by dennisreimann** [reply]
I just verified that in v1.12.5 it works with a plain text like this: `https://mybtcpay.com/i/{Invoice.Id}/receipt`, but not if it's part of a link/inside the `href` attribute.

Will fix it, but for now providing the plaintext version with the `{Invoice.Id}` placeholder is a workaround.

**Reply by dennisreimann** [reply]
Sorry, my bad: It works regardless of how it occurs. I was using the wrong placeholder: It works with `{Invoice.Id}` but not with `{InvoiceId}` — note the dot.

**Reply by Liongrass** [reply]
Great, yes, I can confirm that `https://mybtcpay.com/i/{Invoice.Id}/receipt` works in the hyperlink. Thank you for your help!
